Clone.xs

```diff
@@ -170,7 +170,36 @@ sv_clone (SV * ref, HV* hseen, int depth)
 #endif
       case SVt_PV:		/* 4 */
         TRACEME(("string scalar\n"));
+#if PERL_VERSION >= 20
+        /* only for simple PVs unblessed */
+        if ( SvIsCOW(ref) && !SvOOK(ref) ) {
+          /* cannot use newSVpv_share as this going to use a new PV we do not want to clone it */
+          /* create a fresh new PV */
+          clone = newSV(0);
+          sv_upgrade(clone, SVt_PV);
+          SvPOK_on(clone);
+          SvIsCOW_on(clone);
+          /* points the str slot to the COWed one */
+          SvPV_set(clone, SvPVX(ref) );
+          /* increase the Cow refcnt if possible
+          * Note: we are not doing sv_buf_to_rw + sv_buf_to_ro (mainly required when PERL_DEBUG_READONLY_COW is enabled)
+          * and these APIs are not exported
+          */
+          if ( CowREFCNT(ref) < SV_COW_REFCNT_MAX )
+            CowREFCNT(ref)++;
```

Comment:
This will break (in theory) on certain debug perls because it sets the memory to readonly to see what is trying to scribble on the memory. So you have to unlock the memory before writing to it. It is probably overkill to do this here.

Reply:
probably need to add a check for
```
if ! defined(PERL_DEBUG_READONLY_COW)
```